test/server/clientlib.py

```diff
@@ -7,58 +7,34 @@
 import uuid
 import random
 import time
-from pglib import attach_device
+from pglib import attach_device, get_device_id, get_account_cookie_detail
 
 example_images = {}
 execfile(os.path.normpath(os.path.join(__file__, "../../../bin/load_test_exercise_images.py")), example_images)
 example_images = example_images["example_images"]
 
 
-class ScreenshotsClient(object):
-
+class ScreenshotsBase(object):
     def __init__(self, backend="http://localhost:10080"):
         self.backend = backend
-        self.deviceInfo = make_device_info()
-        self.deviceId = make_uuid()
         self.secret = make_uuid()
-        self.accountId = make_random_id()
         self.session = requests.Session()
         self.session.headers.update({'Accept-Language': 'en-US'})
 
-    def login(self):
-        resp = self.session.post(
-            urljoin(self.backend, "/api/login"),
-            data=dict(deviceId=self.deviceId, secret=self.secret, deviceInfo=json.dumps(self.deviceInfo)))
-        if resp.status_code == 404:
-            resp = self.session.post(
-                urljoin(self.backend, "/api/register"),
-                data=dict(deviceId=self.deviceId, secret=self.secret, deviceInfo=json.dumps(self.deviceInfo)))
-            account_info = attach_device(self.deviceId, self.accountId)
+    def login(self, account_id):
+        # Logs the session in by first checking if a device is
+        # assosciated with account_id and updating session with account cookie
+        device_id = get_device_id(account_id)
+        if device_id:
+            account_info = get_account_cookie_detail(account_id)
             self.session.cookies.update(account_info)
-        resp.raise_for_status()
-        return resp
-
-    def delete_account(self):
-        page = self.session.get(self.backend + "/leave-screenshots/").text
-        csrf_match = re.search(r'<input.*name="_csrf".*value="([^"]*)"', page)
-        csrf = csrf_match.group(1)
-        resp = self.session.post(
-            urljoin(self.backend, "/leave-screenshots/leave"),
-            json={"_csrf": csrf})
-        resp.raise_for_status()
+        return device_id
 
-    def create_shot(self, shot_id=None, **example_args):
-        if not shot_id:
-            shot_id = make_random_id() + "/test.com"
-        shot_url = urljoin(self.backend, shot_id)
-        shot_data = urljoin(self.backend, "data/" + shot_id)
-        shot_json = make_example_shot(self.deviceId, **example_args)
-        resp = self.session.put(
-            shot_data,
-            json=shot_json,
-        )
-        resp.raise_for_status()
-        return shot_url
+    def _get_id_from_url(self, url):
+        assert url.startswith(self.backend)
+        id = url[len(self.backend):]
+        id = id.strip('/')
+        return id
```

Comment:
_get_id_from_url, read_shot, delete_shot, set_expiration are moved from ScreenshotsClient to base class ScreenshotsBase without any change